src/util/debug.h

```diff
@@ -50,6 +50,7 @@ enum sss_logger_t {
 
 extern const char *sss_logger_str[]; /* mapping: sss_logger_t -> string */
 extern const char *debug_prg_name;
+extern bool debug_from_responder;
```

Comment:
This is only to be able to print "CID" instead of "RID"?
Now I'm not so sure in a choice...

Reply:
That's correct, I did not see any other way in debug.c to determine if the message was sent from a responder, or backend/child

Reply:
Strictly speaking, you can infer it from `debug_prg_name`. It might be better so set the from in sss_chain_id_setup()?

```c
uint64_t debug_chain_id;
const char * debug_chain_id_fmt;
```

```c
void sss_chain_id_setup(struct tevent_context *ev, const char *fmt);
```

Reply:
Moving the sss_chain_id_setup() call from `sss_process_init` to `server_setup` (from the other review item) means that `debug_from_responder` value has not been overridden/updated by responders yet when sss_chain_id_setup() is reached. Would a proper solution be adding a new function like sss_chain_id_set_fmt that can be called later?

originally this PR was setting debug_from_responder based on `debug_prg_name` but it was suggested that would likely be too fragile.

Reply:
Just to clarify things, there are conflicting review items here, Alexey suggested to move sss_chain_id_setup() to server_setup(), but then we cannot set the chain ID correct format when calling sss_chain_id_setup() because we don't know we are a responder until sss_process_init() is run (which happens after server_setup)

Reply:
Another thing worth mentioning, we need to setup `fmt` from child processes as well, where we don't call `sss_chain_id_setup()`, I think.

Reply:
So perhaps we can keep sss_chain_id_setup() and add sss_chain_id_set_format()? The format is known in server_setup() you just need to pass it as a parameter.

Reply:
Why? `server_setup` is called from each process' `main`. You can pass the format as additional argument, similar how we already pass the process name.

Reply:
Okay I understand now, thanks for clarification.

Reply:
This is fixed now, child processes are calling `sss_chain_id_set_format` to set the format